### PR TITLE
Add examples for searching repositories (#874)

### PR DESCRIPTION
**Pull request by vishalsodani**

### CODE REVIEW

File: doc/examples/MainClass.rst [hunk 1]
```diff
@@ -37,4 +37,36 @@ Get organization by name
...
+Search repositories by language
+-------------------------------
+
+.. code-block:: python
+
+    >>> repositories = g.search_repositories(query='language:python')
+    >>> for repo in repositories:
+    ...    print(repo.html_url)
+    ...
+    https://github.com/vinta/awesome-python
```
[the remark this answers is not in the document]
Comment (vishalsodani) [reply doc/examples/MainClass.rst]: @sfdye ok. My thinking was that as a user I would want to have the full repo urls.

[the remark this answers is not in the document]
Comment (sfdye) [reply doc/examples/MainClass.rst]: Got it. As a getting-started guide, we try to make it as generic as possible.

[the remark this answers is not in the document]
Comment (vishalsodani) [reply doc/examples/MainClass.rst]: ok

[the remark this answers is not in the document]
Comment (vishalsodani) [reply doc/examples/MainClass.rst]: @sfdye Done.

**Pull request merged**